network/roundrobin.go

```diff
@@ -17,10 +17,13 @@ func NewRoundRobin(server *Server) *RoundRobin {
 }
 
 func (r *RoundRobin) NextProxy(_ IConnWrapper) (IProxy, *gerr.GatewayDError) {
-	proxiesLen := uint32(len(r.proxies))
-	if proxiesLen == 0 {
-		return nil, gerr.ErrNoProxiesAvailable.Wrap(errors.New("proxy list is empty"))
+	if len(r.proxies) > math.MaxUint32 {
+		// This should never happen, but if it does, we fall back to the first proxy.
+		return r.proxies[0], nil
+	} else if len(r.proxies) == 0 {
+		return nil, gerr.ErrNoProxiesAvailable
 	}
+
 	nextIndex := r.next.Add(1)
-	return r.proxies[nextIndex%proxiesLen], nil
+	return r.proxies[nextIndex%uint32(len(r.proxies))], nil //nolint:gosec
```

Comment:
@sinadarbouy `gosec` was complaining about this with this error: `G115: integer overflow conversion int -> uint32`. I fixed it this way. WDYT?

Note that I am ignoring the linter error on line 28, because I am checking it on line 20 already.

Reply:
The overflow is handled by Uint32 itself, and we have a test case to verify this at [roundrobin_test.go, line 86](https://github.com/gatewayd-io/gatewayd/blob/main/network/roundrobin_test.go#L86).
but if we want to avoid linting errors, it’s a good idea to fix the code and double-check it.
LGTM 👍